### ISSUE TITLE
Support for literal Haskell

### ISSUE THREAD

**Issue by Alecton4** [tweag/ormolu#1081]
**Is your feature request related to a problem? Please describe.**
Looks like ormolu does not support literal Haskell, see [details](https://github.com/haskell/haskell-language-server/issues/3912).

**Describe the solution you'd like**
Able to format literal Haskell.

**Describe alternatives you've considered**
N/A

**Additional context**
I'm new to Haskell so not familiar with the tooling. I guess there's certain reasons we do not support literal Haskell? If so, are there other tools can be used?

**Comment by amesgen** [comment]
Thanks for the suggestion! I think implementing support for Literate Haskell should be feasible (ie as supported by the [`unlit`](https://gitlab.haskell.org/ghc/ghc/-/blob/master/utils/unlit/unlit.c) program shipped with GHC, such that eg

```lhs
# A Literate Haskell File

Some imports

> import A
> import B

Very interesting code:

> data A   =   B
```
is formatted to
```lhs
# A Literate Haskell File

Some imports

> import B
> import A

Very interesting code:

> data A = B
```
Is that what you are having in mind?

**Comment by Alecton4** [comment]
@amesgen Yes! Formatting the code parts in literate Haskell is exactly what I mean, sorry for not clearly specifying it. BTW, may I ask the reason why this snippet 

```lhs
> import A
> import B
```

should be formatted to this one?

```lhs
> import B
> import A
```

**Comment by amesgen** [comment]
Ah, thanks for pointing that out, that was just a mistake, it should be exactly reversed, I edited the comment :+1:
